src/qs8-packw/gen/qs8-packw-x8c8-gemm-goi-wasmrelaxedsimd.c

```diff
@@ -0,0 +1,354 @@
+        v128_t v01_1 = wasm_i64x2_shuffle(v0_01, v1_01, 1, 3);
+        v128_t v23_0 = wasm_i64x2_shuffle(v2_01, v3_01, 0, 2);
+        v128_t v23_1 = wasm_i64x2_shuffle(v2_01, v3_01, 1, 3);
+        v128_t v45_0 = wasm_i64x2_shuffle(v4_01, v5_01, 0, 2);
+        v128_t v45_1 = wasm_i64x2_shuffle(v4_01, v5_01, 1, 3);
+        v128_t v67_0 = wasm_i64x2_shuffle(v6_01, v7_01, 0, 2);
+        v128_t v67_1 = wasm_i64x2_shuffle(v6_01, v7_01, 1, 3);
+
+        vacc01 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v01_0, vone, vacc01);
+        vacc01 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v01_1, vone, vacc01);
+        vacc23 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v23_0, vone, vacc23);
+        vacc23 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v23_1, vone, vacc23);
+        vacc45 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v45_0, vone, vacc45);
+        vacc45 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v45_1, vone, vacc45);
+        vacc67 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v67_0, vone, vacc67);
+        vacc67 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v67_1, vone, vacc67);
+
+        wasm_v128_store(out + 0, v01_0);
+        wasm_v128_store(out + 16, v23_0);
+        wasm_v128_store(out + 32, v45_0);
+        wasm_v128_store(out + 48, v67_0);
+
+        wasm_v128_store(out + 64, v01_1);
+        wasm_v128_store(out + 80, v23_1);
+        wasm_v128_store(out + 96, v45_1);
+        wasm_v128_store(out + 112, v67_1);
+
+        w0 += 16;
+        w1 += 16;
+        w2 += 16;
+        w3 += 16;
+        w4 += 16;
+        w5 += 16;
+        w6 += 16;
+        w7 += 16;
+        out += 128;
+      }
+
+      for (; k >= 8; k -= 8) {
+        v128_t v01 = wasm_v128_load64_splat(w0);
+        v01 = wasm_i64x2_shuffle(v01, wasm_v128_load64_splat(w1), 0, 3);
+        v128_t v23 = wasm_v128_load64_splat(w2);
+        v23 = wasm_i64x2_shuffle(v23, wasm_v128_load64_splat(w3), 0, 3);
+        v128_t v45 = wasm_v128_load64_splat(w4);
+        v45 = wasm_i64x2_shuffle(v45, wasm_v128_load64_splat(w5), 0, 3);
+        v128_t v67 = wasm_v128_load64_splat(w6);
+        v67 = wasm_i64x2_shuffle(v67, wasm_v128_load64_splat(w7), 0, 3);
+
+        vacc01 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v01, vone, vacc01);
+        vacc23 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v23, vone, vacc23);
+        vacc45 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v45, vone, vacc45);
+        vacc67 = wasm_i32x4_relaxed_dot_i8x16_i7x16_add(v67, vone, vacc67);
+
+        wasm_v128_store(out + 0, v01);
+        wasm_v128_store(out + 16, v23);
+        wasm_v128_store(out + 32, v45);
+        wasm_v128_store(out + 48, v67);
+
+        w0 += 8;
+        w1 += 8;
+        w2 += 8;
+        w3 += 8;
+        w4 += 8;
+        w5 += 8;
+        w6 += 8;
+        w7 += 8;
+        out += 64;
+      }
+
+      // KC remainder 1..KR-1
+      if (k != 0) {
+        assert(k >= 1 && k <= 7);
+
+        const v128_t vmask = wasm_u64x2_shr(wasm_i32x4_splat(-1), (8 - k) * sizeof(int8_t) * 8);
```

Comment:
nice... i've always meant to try this technique.
its similar to how avx512 would generate a k mask
there is also masked loads for avx512, but only for float, but byte.
this shift technique only works up to 8 bytes I think.  most cpus dont natively have a bit shift of 128 bits or more... after that it becomes a byte shuffle, but the shuffler is usually a vector, so this approach starts to fall down.

This is so much simplier than the avx code that does inserts of 4, 2 and 1 bytes.
For future 4 bit, I think a shuffle is also needed.
e.g.  read 6 bytes, but move them to the first 3 bytes, a pad, and then next 3 bytes.